testapi.pm

```diff
@@ -275,7 +275,15 @@ sub _check_backend_response {
             }
         }
         if (!$check && !$rsp->{saveresult}) {
-            OpenQA::Exception::FailedNeedle->throw(error => "no candidate needle with tag(s) '$mustmatch' matched", tags => $mustmatch);
+            # Must match can be only scalar or array ref.
+            my $needletags = $mustmatch;
+            if (ref($mustmatch)) {
```

Comment:
Not sure about what $mustmatch should be here, but i'd check strictly if it's an [array ref](https://perldoc.perl.org/functions/ref.html) before dereferencing it as an array. `ref(something) eq 'ARRAY'`